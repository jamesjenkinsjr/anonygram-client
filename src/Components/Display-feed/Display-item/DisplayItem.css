.display-item {
    list-style-type: none;
}

.display-img {
    width: 100%;
    min-width: 310px;
    height: auto;
}

.upvote-wrapper {
    position: relative;
    display: flex;
    direction: row;
    justify-content: center;
}

.upvote-count {
    position: absolute;
    bottom: 5px;
    right: 13px;
    font-size: 14px;
    font-weight: bold;
}

.upvote-button {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 30px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 50px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}

.upvote-button:hover,
.upvote-button:focus {
    opacity: 1;
    cursor: pointer;
}

.upvote-button-no-auth {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 30px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 30px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}


.delete-wrapper {
    position: relative;
    display: flex;
    direction: row;
    justify-content: center;
    align-items: center;
}

.delete-button {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 80px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 40px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}

.upvote-button:hover,
.upvote-button:focus {
    opacity: 1;
    cursor: pointer;
}