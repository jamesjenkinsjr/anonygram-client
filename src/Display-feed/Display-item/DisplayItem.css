.display-item {
    list-style-type: none;
}

.display-img {
    width: 100%;
    min-width: 310px;
}

.upvote-wrapper {
    position: relative;
    display: flex;
    direction: row;
    justify-content: center;
}

.upvote-count {
    position: absolute;
    bottom: 10px;
    right: 8px;
    font-size: 14px;
}

.upvote-button {
    position: absolute;
    bottom: 8px;
    right: 30px;
    border: none;
    width: 20px;
    height: 20px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
    margin-right: 10px;
}

.upvote-button:hover,
.upvote-button:focus {
    opacity: 1;
}