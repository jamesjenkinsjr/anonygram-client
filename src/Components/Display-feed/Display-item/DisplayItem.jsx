import React from 'react';
import { Link } from 'react-router-dom';
import './DisplayItem.css';
import { KeyboardArrowUp } from '@material-ui/icons';
import TokenService from '../../../services/token-service';

export default function DisplayItem(props) {
  const { imgAddress, imgCaption, upvotes, id, incrementUpvotes } = props;
  return (
    <li className="display-item">
      <Link to={`/p/${id}`}>
        <img className="display-img" src={imgAddress} alt="anonygram" />
      </Link>
      {TokenService.hasAuthToken() ? <div className="upvote-wrapper">
        <div className="upvote-button">
        <KeyboardArrowUp fontSize="large" onClick={() => incrementUpvotes(id)}/>
        <p className="upvote-count">{upvotes}</p>
        </div>
      </div> 
      :
      <div className="upvote-wrapper">
      <p className="upvote-button-no-auth">{upvotes}</p>
    </div> 
      }
    </li>
  );
}
