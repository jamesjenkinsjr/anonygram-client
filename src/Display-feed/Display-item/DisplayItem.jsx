import React from 'react';
import './DisplayItem.css';

export default function DisplayItem(props) {
    const { imgAddress, upvotes, incrementUpvotes } = props;
    return (
        <li className="display-item">
            <img className="display-img" src={imgAddress} alt="anonygram" />
            <div className="upvote-wrapper">
                <button className="upvote-button" type="button" onClick={() => incrementUpvotes()}>^</button>
                <p className="upvote-count">{upvotes}</p>
            </div>
        </li>
    )
}