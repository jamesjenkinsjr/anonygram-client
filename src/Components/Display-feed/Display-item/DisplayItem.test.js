import React from 'react';
import ReactDOM from 'react-dom';
import DisplayItem from './DisplayItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter} from "react-router-dom";

describe('DisplayItem component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><DisplayItem imageAddress={'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRc53S16cGD_TTr80tPoxgIv5lsF3t52XjOuHV4Rognqp5qBdtO8A&s'} upvotes='20' /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders elements by default', () => {
    const wrapper = shallow(<DisplayItem/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it('renders elements given props', () => {
    const props = {
      imgAddress: 'https://lh3.googleusercontent.com/p/AF1QipMJcAXkTwnOj_TWYbMRwygy8ktducIF2kgOqSdW=s1600-w500',
      imgCaption: 'some image caption', 
      upvotes: 12, 
      id: 20, 
      incrementUpvotes: () => {} 
    }
    
    const wrapper = shallow(<DisplayItem { ...props }/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})