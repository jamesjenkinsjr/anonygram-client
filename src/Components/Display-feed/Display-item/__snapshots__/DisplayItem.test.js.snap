// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayItem component renders elements by default 1`] = `
<li
  className="display-item"
>
  <Link
    to="/p/undefined"
  >
    <img
      alt="anonygram"
      className="display-img"
    />
  </Link>
  <div
    className="upvote-wrapper"
  >
    <p
      className="upvote-button-no-auth"
    />
  </div>
</li>
`;

exports[`DisplayItem component renders elements given props 1`] = `
<li
  className="display-item"
>
  <Link
    to="/p/20"
  >
    <img
      alt="anonygram"
      className="display-img"
      src="https://lh3.googleusercontent.com/p/AF1QipMJcAXkTwnOj_TWYbMRwygy8ktducIF2kgOqSdW=s1600-w500"
    />
  </Link>
  <div
    className="upvote-wrapper"
  >
    <p
      className="upvote-button-no-auth"
    >
      12
    </p>
  </div>
  <p>
    some image caption
  </p>
</li>
`;
